rpcs3/rpcs3qt/game_list_frame.cpp

```diff
@@ -1492,63 +1575,168 @@ void game_list_frame::ShowContextMenu(const QPoint &pos)
 			CreateCPUCaches(gameinfo);
 		}
 	});
-	connect(remove_game, &QAction::triggered, this, [this, current_game, gameinfo, cache_base_dir, name]
+	connect(remove_game, &QAction::triggered, this, [this, current_game, gameinfo, cache_base_dir, hdd1_cache_base_dir, name]
 	{
-		if (current_game.path.empty())
+		if (is_game_running(current_game.serial))
 		{
-			game_list_log.fatal("Cannot remove game. Path is empty");
+			QMessageBox::critical(this, tr("Cannot Remove Game"), tr("The PS3 application is still running, it cannot be removed!"));
 			return;
 		}
 
-		if (is_game_running(current_game.serial))
+		const bool is_disc_game = qstr(current_game.category) == cat::cat_disc_game;
+		const bool is_in_games_dir = is_disc_game && Emu.IsPathInsideDir(current_game.path, rpcs3::utils::get_games_dir());
+		const std::string data_sub_dir = GetFirstSubDirBySerial(rpcs3::utils::get_hdd0_dir() + "/game/", current_game.serial);
+
+		// If data path is present (it could be absent for a disc game)
+		const bool has_data_dir = !is_disc_game || !data_sub_dir.empty();
+		// Data path to be removed (if any)
+		const std::string data_dir = !is_disc_game ? current_game.path : (has_data_dir ? rpcs3::utils::get_hdd0_dir() + "/game/" + data_sub_dir : "");
+		QString text = tr("%0 - %1\n").arg(qstr(current_game.serial)).arg(name);
+
+		if (is_disc_game)
 		{
-			QMessageBox::critical(this, tr("Cannot Remove Game"), tr("The PS3 application is still running, it cannot be removed!"));
-			return;
+			text += tr("\nDisc Game Info:\nPath: %0\n").arg(qstr(current_game.path));
+
+			if (current_game.size_on_disk != umax) // If size was properly detected
+			{
+				text += tr("Size: %0\n").arg(gui::utils::format_byte_size(current_game.size_on_disk));
+			}
+		}
+
+		if (has_data_dir)
+		{
+			text += tr("\nData Info:\nPath: %0\n").arg(qstr(data_dir));
+
+			if (const u64 data_size = fs::get_dir_size(data_dir, 1); data_size != umax) // If size was properly detected
+			{
+				text += tr("Size: %0\n").arg(gui::utils::format_byte_size(data_size));
+			}
+		}
+
+		if (fs::device_stat stat{}; fs::statfs(current_game.path, stat))
+		{
+			text += tr("\nCurrent Free Disk Space: %0\n").arg(gui::utils::format_byte_size(stat.avail_free));
+		}
+		
+		if (has_data_dir)
+		{
+			text += tr("\nPermanently remove Data path and selected (optional) contents from drive?\n");
```

Comment:
I think this makes more sense:
```suggestion
			text += tr("\nPermanently remove %0 and selected (optional) contents from drive?\n").arg(is_disc_game ? tr("Game Data") : gameinfo->localized_category);
```